packages/smart/contracts/libraries/ERC20.sol

```diff
@@ -28,19 +29,28 @@ import "./SafeMathUint256.sol";
  * functions have been added to mitigate the well-known issues around setting
  * allowances. See {IERC20-approve}.
  */
-contract ERC20 is IERC20 {
+abstract contract ERC20 is IERC20 {
     using SafeMathUint256 for uint256;
 
-    uint256 public totalSupply;
+    uint256 public totalSupplyVal;
```

Comment:
OpenZepplin have a good example of how to do this in solidity 8. It applies to 7 as well I think'

https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/ERC20.sol